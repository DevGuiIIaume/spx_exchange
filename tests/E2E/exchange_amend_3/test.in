2
[T0] BUY 0 Router 30 500;
[T1] SELL 0 Router 30 750;
[T0] AMEND 0 40 750;
[T1] SELL 1 Router 10 800;
[T1] AMEND 1 5 750;
[T0] CANCEL 0;
[T0] DISCONNECT;
[T1] DISCONNECT;
