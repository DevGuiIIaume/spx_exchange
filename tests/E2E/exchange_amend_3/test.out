[SPX] Starting
[SPX] Trading 2 products: GPU Router
[SPX] Created FIFO /tmp/spx_exchange_0
[SPX] Created FIFO /tmp/spx_trader_0
[SPX] Connected to /tmp/spx_exchange_0
[SPX] Connected to /tmp/spx_trader_0
[SPX] Created FIFO /tmp/spx_exchange_1
[SPX] Created FIFO /tmp/spx_trader_1
[SPX] Connected to /tmp/spx_exchange_1
[SPX] Connected to /tmp/spx_trader_1
[SPX] [T0] Parsing command: <BUY 0 Router 30 500>
[SPX]	--ORDERBOOK--
[SPX]	Product: GPU; Buy levels: 0; Sell levels: 0
[SPX]	Product: Router; Buy levels: 1; Sell levels: 0
[SPX]		BUY 30 @ $500 (1 order)
[SPX]	--POSITIONS--
[SPX]	Trader 0: GPU 0 ($0), Router 0 ($0)
[SPX]	Trader 1: GPU 0 ($0), Router 0 ($0)
[SPX] [T1] Parsing command: <SELL 0 Router 30 750>
[SPX]	--ORDERBOOK--
[SPX]	Product: GPU; Buy levels: 0; Sell levels: 0
[SPX]	Product: Router; Buy levels: 1; Sell levels: 1
[SPX]		SELL 30 @ $750 (1 order)
[SPX]		BUY 30 @ $500 (1 order)
[SPX]	--POSITIONS--
[SPX]	Trader 0: GPU 0 ($0), Router 0 ($0)
[SPX]	Trader 1: GPU 0 ($0), Router 0 ($0)
[SPX] [T0] Parsing command: <AMEND 0 40 750>
[SPX] Match: Order 0 [T1], New Order 0 [T0], value: $22500, fee: $225.
[SPX]	--ORDERBOOK--
[SPX]	Product: GPU; Buy levels: 0; Sell levels: 0
[SPX]	Product: Router; Buy levels: 1; Sell levels: 0
[SPX]		BUY 10 @ $750 (1 order)
[SPX]	--POSITIONS--
[SPX]	Trader 0: GPU 0 ($0), Router 30 ($-22725)
[SPX]	Trader 1: GPU 0 ($0), Router -30 ($22500)
[SPX] [T1] Parsing command: <SELL 1 Router 10 800>
[SPX]	--ORDERBOOK--
[SPX]	Product: GPU; Buy levels: 0; Sell levels: 0
[SPX]	Product: Router; Buy levels: 1; Sell levels: 1
[SPX]		SELL 10 @ $800 (1 order)
[SPX]		BUY 10 @ $750 (1 order)
[SPX]	--POSITIONS--
[SPX]	Trader 0: GPU 0 ($0), Router 30 ($-22725)
[SPX]	Trader 1: GPU 0 ($0), Router -30 ($22500)
[SPX] [T1] Parsing command: <AMEND 1 5 750>
[SPX] Match: Order 0 [T0], New Order 1 [T1], value: $3750, fee: $38.
[SPX]	--ORDERBOOK--
[SPX]	Product: GPU; Buy levels: 0; Sell levels: 0
[SPX]	Product: Router; Buy levels: 1; Sell levels: 0
[SPX]		BUY 5 @ $750 (1 order)
[SPX]	--POSITIONS--
[SPX]	Trader 0: GPU 0 ($0), Router 35 ($-26475)
[SPX]	Trader 1: GPU 0 ($0), Router -35 ($26212)
[SPX] [T0] Parsing command: <CANCEL 0>
[SPX]	--ORDERBOOK--
[SPX]	Product: GPU; Buy levels: 0; Sell levels: 0
[SPX]	Product: Router; Buy levels: 0; Sell levels: 0
[SPX]	--POSITIONS--
[SPX]	Trader 0: GPU 0 ($0), Router 35 ($-26475)
[SPX]	Trader 1: GPU 0 ($0), Router -35 ($26212)
[SPX] Trader 0 disconnected
[SPX] Trader 1 disconnected
[SPX] Trading completed
[SPX] Exchange fees collected: $263
